### PR TITLE
#4 Implement splash screen with animation

### LINKED ISSUE #4: [UI] Add Splash screen
## Why

Add Splash Screen when preloading the application.
 
## Acceptance Criteria
image

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #4 

## What happened 👀
Add splash screen with animation.
 
## Insight 📝
- Doing some fade-out & animated animation basic.
- Reuse the navigation logic like before - but updating the controller logic a bit with the Future type instead of Optional.obs style.
 
## Proof Of Work 📹
When not logged in, show login screen:

https://user-images.githubusercontent.com/13435717/128342489-e05e2a09-c20a-4b43-ba33-755ba9e5a65f.mp4

When logged in, go to home screen:

https://user-images.githubusercontent.com/13435717/128342523-5652b7bf-34fa-44a3-bb4b-74abb17613be.mp4

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/pages/splash/splash_page.dart [hunk 1]
```diff
@@ -1,40 +1,134 @@
...
+class SplashPage extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _SplashPageState();
+}
+
+class _SplashPageState extends State<SplashPage> with TickerProviderStateMixin {
+  bool _isLogoAnimated = false;
+
+  AnimationController _logoOpacityController;
+  AnimationController _backgroundOpacityController;
```
Comment (manh-t) [inline lib/pages/splash/splash_page.dart:20]: Please add `dispose` for these controllers 🙋

Comment (sleepylee) [reply lib/pages/splash/splash_page.dart:20]: thanks, I added to `dispose()` @manh-t :bowtie:

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**